packages/tradershub/src/hooks/__tests__/usePlatformAssets.spec.ts

```diff
@@ -0,0 +1,132 @@
+import { Regulation } from '@/constants';
+import { useExchangeRates, useTradingAccountsList } from '@deriv/api-v2';
+import { renderHook } from '@testing-library/react-hooks';
+import usePlatformAssets from '../usePlatformAssets';
+
+jest.mock('@deriv/api-v2', () => ({
+    useExchangeRates: jest.fn(),
+    useTradingAccountsList: jest.fn(),
+}));
+
+describe('usePlatformAssets', () => {
+    beforeEach(() => {
+        (useTradingAccountsList as jest.Mock).mockReturnValue({
+            data: [
+                { broker: 'MF', currency: 'USD', is_virtual: false, balance: 100 },
+                { broker: 'CR', currency: 'EUR', is_virtual: false, balance: 200 },
+                { broker: 'MF', currency: 'GBP', is_virtual: true, balance: 300 },
+            ],
+            fiat_account: 'USD',
+            isSuccess: true,
+        });
+
+        (useExchangeRates as jest.Mock).mockReturnValue({
+            data: {},
+            getExchangeRate: () => 1,
+            subscribe: jest.fn(),
+            unsubscribe: jest.fn(),
+        });
+    });
+
+    it('should return correct balances and currency for EU regulation', () => {
+        const { result } = renderHook(() => usePlatformAssets(Regulation.EU));
+
+        expect(result.current.demoAccountBalance).toBe(300);
+        expect(result.current.fiatCurrency).toBe('USD');
+        expect(result.current.isSuccess).toBe(true);
+        expect(result.current.realAccounts).toHaveLength(2);
+        expect(result.current.totalRealPlatformBalance).toBe(100);
+    });
+
+    it('should return correct balances and currency for non-EU regulation', () => {
+        const { result } = renderHook(() => usePlatformAssets(Regulation.NonEU));
+
+        expect(result.current.fiatCurrency).toBe('USD');
+        expect(result.current.isSuccess).toBe(true);
+        expect(result.current.realAccounts).toHaveLength(2);
+        expect(result.current.totalRealPlatformBalance).toBe(200);
+    });
+
+    it('should return 0 for demoAccountBalance when demo account has no balance', () => {
+        (useTradingAccountsList as jest.Mock).mockReturnValue({
+            data: [
+                { broker: 'MF', currency: 'USD', is_virtual: false, balance: 100 },
+                { broker: 'CR', currency: 'EUR', is_virtual: false, balance: 200 },
+                { broker: 'MF', currency: 'GBP', is_virtual: true, balance: undefined },
```

Comment:
@thisyahlen-deriv how balance could be undefined? could it be a real scenario?

Reply:
In regards to useQuery, the first instances will be undefined